src/ray/gcs/gcs_server/gcs_actor_manager.cc

```diff
@@ -649,8 +649,13 @@ Status GcsActorManager::CreateActor(const ray::rpc::CreateActorRequest &request,
       request.task_spec(), actor_namespace, actor_state_counter_);
   actor->UpdateState(rpc::ActorTableData::PENDING_CREATION);
   const auto &actor_table_data = actor->GetActorTableData();
-  // Pub this state for dashboard showing.
-  RAY_CHECK_OK(gcs_publisher_->PublishActor(actor_id, actor_table_data, nullptr));
+  RAY_CHECK_OK(gcs_table_storage_->ActorTable().Put(
```

Comment:
We are not persisting the actor state changes before...
So it makes a difference when listing actors including dead jobs, which takes data from the actor table rather than in the memory cache. Since we never update the table, actors that are stuck in "PENDING_CREATION" are never returned. 

Also, we are publishing the entire actor state, we could just do with the delta.

Reply:
good catch. but this also means we should now handle the race condition. E.g., before the callback is executed if DestroyActor is called (e.g., HandleKillActor request comes in), we can have problems. So we should have the logic like this

```
  auto it = registered_actors_.find(actor_id);
  if (it == registered_actors_.end()) {
    RAY_LOG(INFO) << "Actor is already destroyed " << actor_id;
    return;
  }
```

I think it is safe to just return if it is already deleted.

Reply:
Can you also add a unit test for this case?

Reply:
Are you saying we shouldn't publish actor if the actor is removed from registered actors due to a concurrent actor kill?

Reply:
not just publish, but we shouldn't also do scheduling itself (since it is already killed)

Reply:
yeah that's correct!

Reply:
and actually I think the rest of methods should be inside the callback

Reply:
```
  RemoveUnresolvedActor(actor);

  // Update the registered actor as its creation task specification may have changed due
  // to resolved dependencies.
  registered_actors_[actor_id] = actor;
  // Schedule the actor.
  gcs_actor_scheduler_->Schedule(actor);
```

this part